File: src/main/java/factorio/debugger/DAP/messages/response/DAPVariablesResponse.java
package factorio.debugger.DAP.messages.response;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonTypeName;
import factorio.debugger.DAP.messages.DAPAdditionalProperties;
import factorio.debugger.DAP.messages.DAPResponse;
import factorio.debugger.DAP.messages.types.DAPVariable;

@JsonTypeName("variables")
public class DAPVariablesResponse extends DAPResponse {
    @JsonProperty("body")
    public VariablesResponseBody body;

    public static class VariablesResponseBody extends DAPAdditionalProperties {
        /**
         * All (or a range) of variables for the given variable reference.
         */
        @JsonProperty("variables")
        public DAPVariable[] variables;
    }
}

File: src/main/kotlin/factorio/debugger/DAP/messages/responses/DAPVariablesResponse.kt
package factorio.debugger.DAP.messages.responses

import com.fasterxml.jackson.annotation.JsonProperty
import com.fasterxml.jackson.annotation.JsonTypeName
import factorio.debugger.DAP.messages.DAPAdditionalProperties
import factorio.debugger.DAP.messages.types.DAPVariable

@JsonTypeName("variables")
class DAPVariablesResponse : DAPResponse() {
    @JsonProperty("body")
    lateinit var body: VariablesResponseBody

    class VariablesResponseBody : DAPAdditionalProperties() {
        /**
         * All (or a range) of variables for the given variable reference.
         */
        @JsonProperty("variables")
        lateinit var variables: Array<DAPVariable>
    }
}
